daq/gcp.py

```diff
@@ -252,29 +255,32 @@ def register_offenders(self):
             else:
                 LOGGER.info('Ignoring user %s', user_email)
 
+    def _get_json_report(self, runid):
+        doc = runid.reference.collection('test').document('terminate').get().to_dict()
+        report_blob = doc.get('report.json') if doc else None
+        if not report_blob:
+            return None
+        LOGGER.info('Downloading report %s', report_blob)
+        blob = self._bucket.blob(report_blob)
+        return json.loads(str(blob.download_as_string(), 'utf-8'))
+
     def get_reports_from_date_range(self, device: str, start=None, end=None, count=None):
         """Combine test results from reports within a date range"""
         if not self._firestore:
             LOGGER.error('Firestore not initialized.')
             return
         LOGGER.info('Looking for reports...')
-        origins = self._firestore.collection(u'origin').stream()
```

Comment:
I think we need to keep pulling from origins so that all reports for this device are considered.